.pre-commit-config.yaml

```diff
@@ -0,0 +1,19 @@
+repos:
```

Comment:
Can you describe what this is doing?

Reply:
I have added a git pre-commit hook, that will easy the things for local development. So whenever i commit or push something it will run some things on local like go fmt, go mod tidy, lint, golangcilint and other and if everything is ok, then will allow me to commit something.

Its upto a particular person, if they enable the pre-commit hook on their local repo then this feature will be enabled otherwise it will be disabled and will not cause any issue for others.

Reply:
I'd say lets keep these out of the repository for now, folks are free to maintain these locally if they wish.

Reply:
I think it will help others also, because currently CI is not running on each commit. But will remove it for now.